### PR TITLE
shutdown okhttp thread(include websocket connection) when close execWatch

### PR DESCRIPTION
**Pull request by fanhaozzu**
shutdown okhttp thread(include websocket connection) when close execWatch, okhttp thread will remain a minutes if don't this.

### CODE REVIEW

**Comment by fanhaozzu** [comment]
This will close okhttp threads, of  course it will also close the webSocket connection. If we just only close webSocket, the okhttp threads remain running(of course, the okhttp threads and connections that are held will be released automatically if they remain idle.)

**Comment by fanhaozzu** [comment]
@iocanel

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java [hunk 1]
```diff
@@ -95,14 +97,13 @@ public void close() {
...
             }
-        } catch (Throwable t) {
-            throw KubernetesClientException.launderThrowable(t);
-        }
-
-        WebSocket ws = webSocketRef.get();
-        try {
-            if (ws != null) {
-                ws.close(1000, "Closing...");
+            if (client != null) {
```
Comment (fanhaozzu) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java]: i just want send a command to  apiserver to get result, i don't want keep watching, i need to shutdown all the running thread.

ws.close() is not equivalent to calling client.close(), client.close will shutdown okhttp thread.
If just ws.close(), okhttps threads are still alive.Of course the okhttp thread will close automaticlly after a minutes if there is no request.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java [hunk 2]
```diff
@@ -95,14 +98,13 @@ public void close() {
...
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java]: The lines below effectively (evict idle & shutdown dispatcher) close the client.

Comment (fanhaozzu) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java]: i we use evict idle & shutdown dispatcher, i think ws.close is not nessary

Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java]: But that shuts down the dispatcher for the whole client so no more calls can be made using the client following the websocket close. Client must continue to be usable after websocket close.

Comment (fanhaozzu) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java]: yes, so i think we shoud remove the ws.close, let it released automaticly after a minutes

Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java]: Why? The caller had requested the websocket be closed so we should close the websocket but leave the client usable.

**Comment by iocanel** [comment]
I feel that the changes in this pull request, reflect specific use cases and can't be applied in the client.

@fanhaozzu: I am going to close the pull request for now, as the suggested changes are going to cause quite a few issues to existing users. If you want to rework the pull request, feel free to reopen.

**Pull request closed without merging**